library/src/main/java/com/davidstemmer/screenplay/flow/Screenplay.java

```diff
@@ -14,73 +14,94 @@
-        ArrayDeque<Scene> outgoingScenes;
+        ArrayDeque<Scene> animatedScenesIn;
+        ArrayDeque<Scene> animatedScenesOut;
 
         SceneCut.Builder sceneCut = new SceneCut.Builder()
                 .setDirection(direction)
                 .setCallback(callback);
 
         if (direction == Flow.Direction.BACKWARD) {
-            Scene outgoingScene = previousScenes.getLast();
-            if (outgoingScene.isStacking()) {
-                incomingScenes = new ArrayDeque<>();
-            } else {
-                incomingScenes = getLastSceneBlock(nextBackstack);
-            }
-            outgoingScenes = new ArrayDeque<>();
-            outgoingScenes.push(outgoingScene);
+            ArrayDeque<Scene> difference = difference(previousBackstack, nextBackstack);
+            animatedScenesOut = getLastSceneBlock(difference);
+            animatedScenesIn = moveToNewSceneBlock(difference) ?
+                    getLastSceneBlock(nextBackstack) :
+                    new ArrayDeque<Scene>();
         }
         else {
-            // Using forward/replace, there is always one (and only one) incoming scene
-            incomingScenes = new ArrayDeque<>();
-            incomingScenes.push(incomingScene);
-            // Forward/replace to a stacked scene:
-            // No outgoing scenes to animate or tear down
-            if (incomingScene.isStacking() || previousScenes.isEmpty()) {
-                outgoingScenes = new ArrayDeque<>();
-            }
-            // Forward/replace to a non-stacked scene:
-            // Animate and tear down the previous scene block
-            else {
-                outgoingScenes = previousScenes;
-            }
+            ArrayDeque<Scene> difference = difference(nextBackstack, previousBackstack);
+
+            animatedScenesIn = getLastSceneBlock(difference);
+            animatedScenesOut = moveToNewSceneBlock(difference) ?
+                    getLastSceneBlock(previousBackstack) :
+                    new ArrayDeque<Scene>();
 
         }
 
         Scene.Transformer delegatedTransformer = direction == Flow.Direction.BACKWARD ?
-                outgoingScenes.getFirst().getTransformer():
-                incomingScene.getTransformer();
+                animatedScenesOut.iterator().next().getTransformer():
+                animatedScenesIn.iterator().next().getTransformer();
+
+        sceneCut.setIncomingScenes(animatedScenesIn);
+        sceneCut.setOutgoingScenes(animatedScenesOut);
 
-        for (Scene scene : incomingScenes) {
-            sceneCut.addIncomingScene(scene);
+        previousBackstack.clear();
+        previousBackstack.addAll(nextBackstack);
+
+        beginCut(sceneCut.build(), delegatedTransformer);
+    }
+
+    private ArrayDeque<Scene> deque(Backstack backstack) {
+        ArrayDeque<Scene> sceneSet = new ArrayDeque<Scene>();
+        for (Backstack.Entry entry : backstack) {
+            Scene nextScene = (Scene) entry.getScreen();
```

Comment:
Why is there a name difference here? Why does entry have a getScreen method instead of getScene?

/soapbox